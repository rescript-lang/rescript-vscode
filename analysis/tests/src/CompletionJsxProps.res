// let _ = <CompletionSupport.TestComponent on=
//                                             ^com

// let _ = <CompletionSupport.TestComponent on=t
//                                              ^com

// let _ = <CompletionSupport.TestComponent test=T
//                                                ^com

// let _ = <CompletionSupport.TestComponent polyArg=
//                                                  ^com

// let _ = <CompletionSupport.TestComponent polyArg=#t
//                                                    ^com

// let _ = <div muted= />
//                    ^com

// let _ = <div onMouseEnter= />
//                           ^com

// Should wrap in {}
// let _ = <CompletionSupport.TestComponent testArr=
//                                                  ^com

// Should not wrap in {}
// let _ = <CompletionSupport.TestComponent testArr={[]}
//                                                    ^com

let tsomeVar = #two

// let _ = <CompletionSupport.TestComponent polyArg={}
//                                                   ^com

// let _ = <CompletionSupport.TestComponent on={t}
//                                               ^com
