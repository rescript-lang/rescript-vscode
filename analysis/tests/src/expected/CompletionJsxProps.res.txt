Complete src/CompletionJsxProps.res 0:47
posCursor:[0:47] posNoWhite:[0:46] Found expr:[0:12->0:47]
JSX <CompletionSupport.TestComponent:[0:12->0:43] on[0:44->0:46]=...__ghost__[0:-1->0:-1]> _children:None
Completable: Cexpression CJsxPropValue [CompletionSupport, TestComponent] on
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [CompletionSupport, TestComponent] on
Path CompletionSupport.TestComponent.make
[{
    "label": "true",
    "kind": 4,
    "tags": [],
    "detail": "bool",
    "documentation": null
  }, {
    "label": "false",
    "kind": 4,
    "tags": [],
    "detail": "bool",
    "documentation": null
  }]

Complete src/CompletionJsxProps.res 3:48
posCursor:[3:48] posNoWhite:[3:47] Found expr:[3:12->3:48]
JSX <CompletionSupport.TestComponent:[3:12->3:43] on[3:44->3:46]=...[3:47->3:48]> _children:None
Completable: Cexpression CJsxPropValue [CompletionSupport, TestComponent] on=t
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [CompletionSupport, TestComponent] on
Path CompletionSupport.TestComponent.make
[{
    "label": "true",
    "kind": 4,
    "tags": [],
    "detail": "bool",
    "documentation": null
  }]

Complete src/CompletionJsxProps.res 6:50
posCursor:[6:50] posNoWhite:[6:49] Found expr:[6:12->6:50]
JSX <CompletionSupport.TestComponent:[6:12->6:43] test[6:44->6:48]=...[6:49->6:50]> _children:None
Completable: Cexpression CJsxPropValue [CompletionSupport, TestComponent] test=T
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [CompletionSupport, TestComponent] test
Path CompletionSupport.TestComponent.make
[{
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\ntype testVariant = One | Two | Three(int)",
    "documentation": null,
    "sortText": "A Two",
    "insertText": "{Two}",
    "insertTextFormat": 2
  }, {
    "label": "Three(_)",
    "kind": 4,
    "tags": [],
    "detail": "Three(int)\n\ntype testVariant = One | Two | Three(int)",
    "documentation": null,
    "sortText": "A Three(_)",
    "insertText": "{Three($0)}",
    "insertTextFormat": 2
  }, {
    "label": "TableclothMap",
    "kind": 9,
    "tags": [],
    "detail": "file module",
    "documentation": null
  }, {
    "label": "TypeAtPosCompletion",
    "kind": 9,
    "tags": [],
    "detail": "file module",
    "documentation": null
  }, {
    "label": "TypeDefinition",
    "kind": 9,
    "tags": [],
    "detail": "file module",
    "documentation": null
  }]

Complete src/CompletionJsxProps.res 9:52
posCursor:[9:52] posNoWhite:[9:51] Found expr:[9:12->9:52]
JSX <CompletionSupport.TestComponent:[9:12->9:43] polyArg[9:44->9:51]=...__ghost__[0:-1->0:-1]> _children:None
Completable: Cexpression CJsxPropValue [CompletionSupport, TestComponent] polyArg
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [CompletionSupport, TestComponent] polyArg
Path CompletionSupport.TestComponent.make
[{
    "label": "#one",
    "kind": 4,
    "tags": [],
    "detail": "#one\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "{#one}",
    "insertTextFormat": 2
  }, {
    "label": "#three(_, _)",
    "kind": 4,
    "tags": [],
    "detail": "#three(int, bool)\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "{#three(${1:_}, ${2:_})}",
    "insertTextFormat": 2
  }, {
    "label": "#two",
    "kind": 4,
    "tags": [],
    "detail": "#two\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "{#two}",
    "insertTextFormat": 2
  }, {
    "label": "#two2",
    "kind": 4,
    "tags": [],
    "detail": "#two2\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "{#two2}",
    "insertTextFormat": 2
  }]

Complete src/CompletionJsxProps.res 12:54
posCursor:[12:54] posNoWhite:[12:53] Found expr:[12:12->12:54]
JSX <CompletionSupport.TestComponent:[12:12->12:43] polyArg[12:44->12:51]=...[12:52->12:54]> _children:None
Completable: Cexpression CJsxPropValue [CompletionSupport, TestComponent] polyArg=#t
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [CompletionSupport, TestComponent] polyArg
Path CompletionSupport.TestComponent.make
[{
    "label": "#three(_, _)",
    "kind": 4,
    "tags": [],
    "detail": "#three(int, bool)\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "{three(${1:_}, ${2:_})}",
    "insertTextFormat": 2
  }, {
    "label": "#two",
    "kind": 4,
    "tags": [],
    "detail": "#two\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "{two}",
    "insertTextFormat": 2
  }, {
    "label": "#two2",
    "kind": 4,
    "tags": [],
    "detail": "#two2\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "{two2}",
    "insertTextFormat": 2
  }]

Complete src/CompletionJsxProps.res 15:22
posCursor:[15:22] posNoWhite:[15:21] Found expr:[15:12->15:25]
JSX <div:[15:12->15:15] muted[15:16->15:21]=...__ghost__[0:-1->0:-1]> _children:15:23
Completable: Cexpression CJsxPropValue [div] muted
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [div] muted
Path ReactDOM.domProps
Path Pervasives.JsxDOM.domProps
[{
    "label": "true",
    "kind": 4,
    "tags": [],
    "detail": "bool",
    "documentation": null
  }, {
    "label": "false",
    "kind": 4,
    "tags": [],
    "detail": "bool",
    "documentation": null
  }]

Complete src/CompletionJsxProps.res 18:29
posCursor:[18:29] posNoWhite:[18:28] Found expr:[18:12->18:32]
JSX <div:[18:12->18:15] onMouseEnter[18:16->18:28]=...__ghost__[0:-1->0:-1]> _children:18:30
Completable: Cexpression CJsxPropValue [div] onMouseEnter
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [div] onMouseEnter
Path ReactDOM.domProps
Path Pervasives.JsxDOM.domProps
[{
    "label": "event => {}",
    "kind": 12,
    "tags": [],
    "detail": "JsxEventC.Mouse.t => unit",
    "documentation": null,
    "sortText": "A",
    "insertText": "{${1:event} => {$0}}",
    "insertTextFormat": 2
  }]

Complete src/CompletionJsxProps.res 22:52
posCursor:[22:52] posNoWhite:[22:51] Found expr:[22:12->22:52]
JSX <CompletionSupport.TestComponent:[22:12->22:43] testArr[22:44->22:51]=...__ghost__[0:-1->0:-1]> _children:None
Completable: Cexpression CJsxPropValue [CompletionSupport, TestComponent] testArr
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [CompletionSupport, TestComponent] testArr
Path CompletionSupport.TestComponent.make
[{
    "label": "[]",
    "kind": 12,
    "tags": [],
    "detail": "testVariant",
    "documentation": null,
    "sortText": "A",
    "insertText": "{[$0]}",
    "insertTextFormat": 2
  }]

Complete src/CompletionJsxProps.res 26:54
posCursor:[26:54] posNoWhite:[26:53] Found expr:[26:12->26:56]
JSX <CompletionSupport.TestComponent:[26:12->26:43] testArr[26:44->26:51]=...[26:53->26:55]> _children:None
Completable: Cexpression CJsxPropValue [CompletionSupport, TestComponent] testArr->array
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [CompletionSupport, TestComponent] testArr
Path CompletionSupport.TestComponent.make
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\ntype testVariant = One | Two | Three(int)",
    "documentation": null,
    "insertText": "One",
    "insertTextFormat": 2
  }, {
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\ntype testVariant = One | Two | Three(int)",
    "documentation": null,
    "insertText": "Two",
    "insertTextFormat": 2
  }, {
    "label": "Three(_)",
    "kind": 4,
    "tags": [],
    "detail": "Three(int)\n\ntype testVariant = One | Two | Three(int)",
    "documentation": null,
    "insertText": "Three($0)",
    "insertTextFormat": 2
  }]

Complete src/CompletionJsxProps.res 31:53
posCursor:[31:53] posNoWhite:[31:52] Found expr:[31:12->31:54]
JSX <CompletionSupport.TestComponent:[31:12->31:43] polyArg[31:44->31:51]=...[31:52->31:54]> _children:None
Completable: Cexpression CJsxPropValue [CompletionSupport, TestComponent] polyArg->recordBody
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [CompletionSupport, TestComponent] polyArg
Path CompletionSupport.TestComponent.make
[{
    "label": "#one",
    "kind": 4,
    "tags": [],
    "detail": "#one\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "#one",
    "insertTextFormat": 2
  }, {
    "label": "#three(_, _)",
    "kind": 4,
    "tags": [],
    "detail": "#three(int, bool)\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "#three(${1:_}, ${2:_})",
    "insertTextFormat": 2
  }, {
    "label": "#two",
    "kind": 4,
    "tags": [],
    "detail": "#two\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "#two",
    "insertTextFormat": 2
  }, {
    "label": "#two2",
    "kind": 4,
    "tags": [],
    "detail": "#two2\n\n[#one | #three(int, bool) | #two | #two2]",
    "documentation": null,
    "insertText": "#two2",
    "insertTextFormat": 2
  }]

Complete src/CompletionJsxProps.res 34:49
posCursor:[34:49] posNoWhite:[34:48] Found expr:[34:12->34:50]
JSX <CompletionSupport.TestComponent:[34:12->34:43] on[34:44->34:46]=...[34:48->34:49]> _children:None
Completable: Cexpression CJsxPropValue [CompletionSupport, TestComponent] on=t->recordBody
Package opens Pervasives.JsxModules.place holder
Resolved opens 1 pervasives
ContextPath CJsxPropValue [CompletionSupport, TestComponent] on
Path CompletionSupport.TestComponent.make
[{
    "label": "true",
    "kind": 4,
    "tags": [],
    "detail": "bool",
    "documentation": null
  }, {
    "label": "tsomeVar",
    "kind": 12,
    "tags": [],
    "detail": "[> #two]",
    "documentation": null
  }]

